lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -974,6 +980,16 @@ def set_regexp_patterns
 
       @compiled_http_latency_regexp =
         /^\s*(?<seconds>\d+)(?<decimal>\.\d+)?\s*s\s*$/
+
+      # TODO(qingling128): Temporary fallback for metadata agent restarts.
+      @compiled_k8s_container_local_resource_id_regexp = /^
+        (?<resource_type>k8s_container)
+        \.(?<namespace_name>[0-9a-z-]+)
+        \.(?<pod_name>[0-9a-z-.]+)
```

Comment:
I think this is broken. `-` needs to be the first or the last within the `[]` to be treated as a literal `-`. Maybe use `[.0-9a-z-]` instead?